UnityProject/Assets/Scripts/Systems/SmokeAndFoam/SmokeAndFoamManager.cs

```diff
@@ -40,6 +40,8 @@ public static void StartSmokeAt(Vector3 PositionWorld, ReagentMix Container, int
 		SourceReservoir.PerTile.Multiply(SourceReservoir.ReagentPurity);
 		SourceReservoir.StacksLeft = AmountOfSmoke;
 
+		Container.Take(AmountOfSmoke - 1);
```

Comment:
ok, 
theoretically if you want to make something smoke , should it use up all the reagents in a container, like is the entire contents of the container getting Carried in the smoke?

Reply:
SO think about it this way:
I have a beaker, 500 tiles of smoke, 50u of acid. Right now, that smoke will place 0.1u of smoke on everytile, but only conusme 1u of acid from the beaker.

This is a bug fix so that 500 tiles of smoke, will place 0.1u on each tile, consuming 50u of acid in the process,

Reply:
When working on it I don't mind the reagent duplication in the air, 
since with stuff like healing smoke if you get a lot of it to get an actual decent amount of healing you would have to have like 300u and I don't think any containers have that

Reply:
I havent changed the logic of how much reagent is in the cloud at all. I just made it actually use up the reagent it puts in the cloud

Reply:
Also while it wasnt the point you were making, we have bluespace beakers which have 300u i believe

Reply:
yeha, it's fine to consume the entire beaker

Reply:
probably best to make it do that, what about a scenario where it's 1 smoke and 299u acid?
is it just very acidic smoke xD in one tile

Reply:
nah, the current weight works would be it would just be 1u acid on that tile, 
so it just consumes the entire container?

Reply:
If its just one tile... yeah that smoke is just gonna have some incredible concentration lol

Reply:
or should it only consume the equivalent amount of acid one smoke equals one acid consumed?

Reply:
You saying to add like a 'max saturation' to the clouds? I mean could do, but is there any downside to letting one tile of smoke be super dense? It doesnt break anything as at that point its arguably less effective than just throwing the beaker and letting it spill

Reply:
I think it's good